src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Document.cs

```diff
@@ -107,5 +112,42 @@ private void ScanTagInDocumentContext()
                 tagBlock.Dispose();
             }
         }
+
+        private bool CurrentTagBlockExpectsJavascriptContent()
+        {
+            var blockBuilder = Context.CurrentBlock;
+
+            Debug.Assert(blockBuilder != null);
+
+            var potentialAttributes = blockBuilder.Children
+                .OfType<Block>()
+                .Where(block =>
+                    block.ChunkGenerator is AttributeBlockChunkGenerator &&
+                    block.Children.Count() >= 2);
+
+            if (potentialAttributes.Any())
```

Comment:
Write one Linq expression (remove the `potentialAttributes` intermediate) and don't perform this additional evaluation. `FirstOrDefault()` does the `Any()` bit already.